test/integration-test.sh

```diff
@@ -204,44 +183,184 @@ redeem_tickets() {
   last_redeemed="${redeemed}"
 
   # Trigger a redemption run, but cap it at 1 minute. We only want to measure
   # progress, not redeeem all tickets which takes too long.
   log "Node ${node_id} should redeem all tickets"
-  result=$(run_command "${node_api}" "redeemTickets" "" 60 60)
+  # add 60 second timeout 
+  result=$(run_api ${node_api} "/tickets/redeem" "POST" "" "" 60 60)
   log "--${result}"
 
   # Get ticket statistics again and compare with previous state. Ensure we
   # redeemed tickets.
-  result=$(run_command "${node_api}" "tickets" "" 600)
+  result=$(get_tickets_statistics ${node_api} "winProportion")
   log "Node ${node_id} ticket information (after redemption) -- ${result}"
-  rejected=$(echo "${result}" | grep "Rejected:" | awk '{ print $3; }' | tr -d '\n')
-  redeemed=$(echo "${result}" | grep "Redeemed:" | awk '{ print $3; }' | tr -d '\n')
+  rejected=$(echo "${result}" | jq -r .rejected)
+  redeemed=$(echo "${result}" | jq -r .redeemed)
   [[ ${rejected} -gt 0 ]] && { msg "rejected tickets count on node ${node_id} is ${rejected}"; exit 1; }
   [[ ${redeemed} -gt 0 && ${redeemed} -gt ${last_redeemed} ]] || { msg "redeemed tickets count on node ${node_id} is ${redeemed}, previously ${last_redeemed}"; exit 1; }
   last_redeemed="${redeemed}"
 
   # Trigger another redemption run, but cap it at 1 minute. We only want to measure
   # progress, not redeeem all tickets which takes too long.
   log "Node ${node_id} should redeem all tickets (again to ensure re-run of operation)"
-  result=$(run_command "${node_api}" "redeemTickets" "" 60 60)
+  # add 60 second timeout 
+  result=$(run_api ${node_api} "/tickets/redeem" "POST" "" "" 60 60)
   log "--${result}"
 
   # Get final ticket statistics
-  result=$(run_command "${node_api}" "tickets" "" 600)
+  result=$(get_tickets_statistics ${node_api} "winProportion")
   log "Node ${node_id} ticket information (after second redemption) -- ${result}"
-  rejected=$(echo "${result}" | grep "Rejected:" | awk '{ print $3; }' | tr -d '\n')
-  redeemed=$(echo "${result}" | grep "Redeemed:" | awk '{ print $3; }' | tr -d '\n')
+  rejected=$(echo "${result}" | jq -r .rejected)
+  redeemed=$(echo "${result}" | jq -r .redeemed)
   [[ ${rejected} -gt 0 ]] && { msg "rejected tickets count on node ${node_id} is ${rejected}"; exit 1; }
   [[ ${redeemed} -gt 0 && ${redeemed} -gt ${last_redeemed} ]] || { msg "redeemed tickets count on node ${node_id} is ${redeemed}, previously ${last_redeemed}"; exit 1; }
 }
 
+# $1 = node api endpoint
+# $2 = currency to withdraw
+# $3 = amount to withdraw
+# $4 = where to send the funds to
+withdraw() {
+  local node_api="${1}"
+  local currency="${2}"
+  local amount="${3}"
+  local recipient="${4}"
+
+  result=$(run_api ${node_api} "/account/withdraw" "POST" "{\"currency\": \"${currency}\", \"amount\": \"${amount}\", \"recipient\": \"${recipient}\"}" "receipt" 600)
+  echo "${result}"
```

Comment:
If you are echoing the result straight up, you may just run the function without storing the result.

This applies to all the other helper function following this one.